src/main/java/com/drajer/bsa/kar/action/SubmitReport.java

```diff
@@ -14,34 +17,45 @@
 import org.hl7.fhir.r4.model.Resource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class SubmitReport extends BsaAction {
 
   private final Logger logger = LoggerFactory.getLogger(SubmitReport.class);
-
+  private static final String R4 = "R4";
   private String submissionEndpoint;
 
-  private static final FhirContext context = FhirContext.forR4();
+  /** The FHIR Context Initializer necessary to retrieve FHIR resources */
+  @Autowired FhirContextInitializer fhirContextInitializer;
+
+  @Autowired PublicHealthAuthorityService publicHealthAuthorityService;
 
   @Override
   public BsaActionStatus process(KarProcessingData data, EhrQueryService ehrService) {
 
     logger.info(" Executing the submission of the Report to : {}", submissionEndpoint);
+    FhirContext context = fhirContextInitializer.getFhirContext(R4);
 
     BsaActionStatus actStatus = new SubmitReportStatus();
     actStatus.setActionId(this.getActionId());
-
     // Check Timing constraints and handle them before we evaluate conditions.
     BsaActionStatusType status = processTimingData(data);
-
+    PublicHealthAuthority pha =
+        publicHealthAuthorityService.getPublicHealthAuthorityByUrl(submissionEndpoint);
```

Comment:
The only thing I am a little worried about here is that the submission endpoint will point to a `$process-message`, which we would have to make sure we add to the URL when we register the server. Could we add a quick little check, if we don't find an exact match, and the submission endpoint doesn't contain `$process-message`, try adding it and getting by URL again?